cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -153,6 +153,26 @@ int64_t kafka_consumer::get_committed_offset(std::string const &topic, int parti
   return offset > 0 ? offset : 0;
 }
 
+const std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics()
+{
+  RdKafka::Metadata *md;
+  std::map<std::string, std::vector<int32_t>> topic_parts;
+
+  CUDF_EXPECTS(RdKafka::ERR_NO_ERROR == consumer->metadata(true, nullptr, &md, default_timeout),
+               "Failed to list_topics in Kafka broker");
+
+  const RdKafka::Metadata::TopicMetadataVector *mdv = md->topics();
+
+  for (auto &topic : *mdv) {
+    std::vector<int32_t> t_parts;
+    for (auto &partition : *(topic->partitions())) { t_parts.push_back(partition->id()); }
+    topic_parts[topic->topic()] = t_parts;
+  }
+
+  delete md, mdv;
```

Comment:
Pretty sure mdv is just a pointer to something owned by md. Are you sure it's correct to delete it?

Reply:
Not entirely sure to be honest @harrism? I completely agree with and see your point but hte documentation makes it clear to call "delete" so to be honest. I'm not sure. Any ideas?

Reply:
I didn't know this was calling into an external library. If it says delete, do what it says I guess...  rdkafka API looks a bit old-school.